### ISSUE TITLE
Updating learning rate with Libtorch 1.5 and optimiser options

### ISSUE THREAD

**Issue by jmorrill1** [pytorch/pytorch#40098]
This is my first question ever on Github, so please accept my apologies if I have not posted this in the right place. With the release of the 1.5 stable version of the C++ API for PyTorch, there are some changes in some of the object interfaces. For instance, now

`optimizer.options.learning_rate();`

won't work (here the optimiser being used is Adam) since `learning_rate` has changed to `lr` (see https://github.com/pytorch/pytorch/releases) but moreover the optimiser no longer has options (`no member named 'options' in 'torch::optim::Adam'`). So my question is: how would one run

`optimizer.options.learning_rate();`

or update the learning rate

`optimizer.options.learning_rate(updatedlearningrate);`

with the new release? Any help will be appreciated! Thank you

**Comment by vincentqb** [comment]
cc @yf225 and @anjali411 from #29335

**Comment by anjali411** [comment]
If you don't set explicit options for each ParamGroup, all ParamGroups will have common options stored in `defaults` in `Optimizer` class. you can obtain any entry in defaults by static casting it to the respective `OptimizerOptions`.
In this case:
```
auto options = static_cast<AdamOptions&>(optimizer.defaults());
// access learning rate
auto lr = options.lr();
// update learning rate
options.lr(updatedlearningrate)
```

**Comment by aportelli** [comment]
Thanks! That's very useful (I am working with @jmorrill1).
Another thing: I could not get the option calls type in a way which is independent from the optimiser type. I made a type trait

```cpp
template <typename T>
struct OptimizerOptions {};

template <>
struct OptimizerOptions<torch::optim::Adam>
{
  typedef torch::optim::AdamOptions type;
};
```

which could be populated for other optimiser, but somewhat it would be more natural if we could access the type with something like `torch::optim::Adam::Options`, with the convention that `::Options` would be here for each optimiser. Is it already here and I just missed it?
